src/arch/amdgpu/vega/insts/vop3_cvt.hh

```diff
@@ -547,26 +549,23 @@ execute(GPUDynInstPtr gpuDynInst) override
     // The gem5 operand types are really only handy up to 64 bits. For BF
     // operand sizes such as in these instructions, just create an array of
     // 32-bit registers to use.
-    typename std::aligned_storage<sizeof(ConstVecOperandU32),
-                                  alignof(ConstVecOperandU32)>::type
-                                      _src0[input_regs];
-    ConstVecOperandU32 *src0 =
-        std::launder(reinterpret_cast<ConstVecOperandU32*>(&_src0));
+    std::vector<ConstVecOperandU32> src0;
+    src0.reserve(input_regs);
     for (int reg = 0; reg < input_regs; ++reg) {
-        new (&src0[reg]) ConstVecOperandU32(gpuDynInst, extData.SRC0 + reg);
+        src0.emplace_back(gpuDynInst, extData.SRC0 + reg);
         src0[reg].readSrc();
     }
 
     ConstVecOperandF32 src1(gpuDynInst, extData.SRC1);
     src1.readSrc();
 
-    typename std::aligned_storage<sizeof(VecOperandU32),
-                                  alignof(VecOperandU32)>::type
-                                      _vdst[output_regs];
-    VecOperandU32 *vdst =
-        std::launder(reinterpret_cast<VecOperandU32*>(&_vdst));
+    std::vector<typename std::aligned_storage<sizeof(VecOperandU32),
```

Comment:
I'm a little confused about this change -- as far as I can tell, _vdst becoming a vector does not seem to impact the subsequent code -- maybe I'm missing something subtle?

Maybe calling data() on it iterates through the vector?

Reply:
I'm not sure I fully understand what the compiler didn't like about the array, since the declaration seems fine to me, but given your explanation the updated code also doesn't seem wrong so I'm fine with it